pandas/core/internals/blocks.py

```diff
@@ -3490,6 +3490,7 @@ def _putmask_smart(v, m, n):
 
         # we ignore ComplexWarning here
         with warnings.catch_warnings(record=True):
+            warnings.simplefilter("ignore", np.ComplexWarning)
```

Comment:
does the context manager restore all of the warnings filters after? (I guess that is the point?)

Reply:
Yeah, that's the idea. Ideally libraries shouldn't modify the global warnings registry without good cause.